include/grpcpp/generic/generic_stub_impl.h

```diff
@@ -38,44 +39,56 @@ typedef ::grpc_impl::ClientAsyncResponseReader<ByteBuffer>
 namespace grpc_impl {
 class CompletionQueue;
 
-/// Generic stubs provide a type-unsafe interface to call gRPC methods
+/// Generic stubs provide a type-unaware interface to call gRPC methods
 /// by name.
-class GenericStub final {
+template <class MessageType>
```

Comment:
Can there be use cases where the request and response MessageTypes might be different for a generic call?

Reply:
Practically, an asymmetric use case is not very useful. Most likely you'll have a common base class (like `proto2::MessageLite`) at the base of your serialization hierarchy and thus can use this for at least basic ser/deser. Maybe the case where that wouldn't work is if you're using a service to translate between data formats, but that doesn't seem immediately useful (and you can always use raw ByteBuffer in that case).

Theoretically, there could be a use at some point, but I'm trying to keep this customer-driven (esp since this is just within the grpc_impl namespace, not true API) and we haven't had a request for anything other than symmetric use at the current time.

Reply:
I'm going to go ahead and make the change that you suggested. It's fine to do so in the `impl` namespace even if it never makes it to top-level API.